pulp_service/pulp_service/app/tasks/package_scan.py

```diff
@@ -0,0 +1,88 @@
+from pulp_npm.app.models import Package as NPMPackage
+from pulp_service.app.constants import PKG_ECOSYSTEM, OSV_QUERY_URL
+from pulp_service.app.models import VulnerabilityReport
+
+# Create a thread-safe queue to share Content units between threads
+content_queue = Queue()
+
+
+async def check_content(repo_version_pk):
+    """
+    Get the list of contents from reop_version, build a package_list and makes an API request to
+    osv.dev using this package_list.
+    """
+    # create a thread to collect the Contents from RepositoryVersion
+    await asyncio.to_thread(_get_content_from_repo, repo_version_pk)
+    await _scan_packages()
+
+
+async def _scan_packages():
+    """
+    Makes a request to the osv.dev API and store the results in VulnerabilityReport model.
+    """
+    scanned_packages = {}
+    async with aiohttp.ClientSession() as session:
+        for package in _define_osv_package_list():
+            osv_data = json.dumps(package["osv_data"])
+            async with session.post(url=OSV_QUERY_URL, data=osv_data) as response:
+                response_body = await response.text()
+                json_body = json.loads(response_body)
+                if json_body.get("vulns"):
+                    package_name = "{package}-{version}".format(
+                        package=package["osv_data"]["package"]["name"],
+                        version=package["osv_data"]["version"],
+                    )
+                    scanned_packages[package_name] = json_body["vulns"]
+    await sync_to_async(VulnerabilityReport.objects.get_or_create)(
+        vulns=scanned_packages,
+    )
+
+
+def _get_content_from_repo(repo_version_pk: str):
+    """
+    Populate content_queue Queue with the content_units found in RepositoryVersion
+    """
+    repo_version = RepositoryVersion.objects.get(pk=repo_version_pk)
+    for content_unit in repo_version.content:
+        content_queue.put(content_unit.cast())
+    content_queue.put(None)  # signal that there is no more content_units
+
+
+def _define_osv_package_list():
+    """
+    Retrieves the content_units from Queue and, for each content found, yields a dicitonary
+    following the osv.dev expected format
+    { "package": {"name": "<package name>", "ecosystem": "<ecosystem>" }, "version": "<version>" }
+    """
+    while True:
+        content = content_queue.get()
+        if not content:  # if there is nothing else in the queue, break execution
+            break
+        ecosystem = _identify_package_ecosystem(content)
+        if not ecosystem:
+            break
+        yield {
+            "osv_data": {
+                "package": {"name": content.name, "ecosystem": ecosystem},
+                "version": content.version,
+            },
+            "content": content,
+        }
+
+
+def _identify_package_ecosystem(content: any) -> str:
+    """
+    Returns an osv.dev ecosystem (string) based on the content_type
+    """
+    model_type = content.TYPE
```

Comment:
Consider removing this default value since we're going to want to explicitly add osv.dev support type-by-type later.

Instead I'd remove this line and add an `else` so that if none of the "supported content types" (in this case just NPMPackage) are not matched it raises a Python exception. Basically don't let unsupported types pass silently I think would be good.